### DISCUSSION TITLE
XAML conditional prefixes mobile and desktop

### DISCUSSION THREAD

**Discussion by workgroupengineering** [unoplatform/uno#7922] [Q&A] [answered]
## What would you like to be added:
Add new XAML conditional prefixes to identify mobile  and desktop platform
## Why is this needed:
Some desktop controls are not usable from mobile (DataGrid) and vice versa. To solve this case it generally uses StateTrigger. With these prefixes it is not necessary to use StateTrigger and will also reduce the memory allocation.
## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [x] Build tasks
- [ ] Solution Templates

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the suggestion!

In general, what's considered mobile and desktop is fuzzy and depends on the application. You can create your own namespace exclusions [using this pattern](https://github.com/unoplatform/Uno.Themes/blob/7b96c64c4aef12ccf3faebe90dde5e37d8cd3a0a/src/library/Uno.Material/Uno.Material.csproj#L23-L36):

```xaml
	<Choose>
		<When Condition="'$(TargetFramework)'=='xamarinios10' or '$(TargetFramework)'=='monoandroid10.0' or '$(TargetFramework)'=='monoandroid11.0'">
			<ItemGroup>
				<IncludeXamlNamespaces Include="mobile" />
				<ExcludeXamlNamespaces Include="not_mobile" />
			</ItemGroup>
		</When>
		<Otherwise>
			<ItemGroup>
				<IncludeXamlNamespaces Include="not_mobile" />
				<ExcludeXamlNamespaces Include="mobile" />
			</ItemGroup>
		</Otherwise>
	</Choose>
```
